draft-ietf-lake-edhoc.md

```diff
@@ -711,42 +708,74 @@ The definition of Expand depends on the EDHOC hash algorithm of the selected cip
+IV_4          = EDHOC-KDF( PRK_4e3m, 9, TH_4,      iv_length )
+~~~~~~~~~~~~~~~~~~~~~~~
+{: #fig-edhoc-kdf title="Key derivations using EDHOC-KDF"}
+{: artwork-align="center"}
+
+The pseudo-random key PRK_out is used to derive application specific data and for key update.
 
-* KEYSTREAM_2 is derived using the transcript hash TH_2 and the pseudorandom key PRK_2e.
-* MAC_2 is derived using the transcript hash TH_2 and the pseudorandom key PRK_3e2m.
-* K_3 and IV_3 are derived using the transcript hash TH_3 and the pseudorandom key PRK_3e2m. IVs are only used if the EDHOC AEAD algorithm uses IVs.
-* MAC_3 is derived using the transcript hash TH_3 and the pseudorandom key PRK_3m.
 
-KEYSTREAM_2, K_3, and IV_3 use an empty CBOR byte string h'' as context. MAC_2 and MAC_3 use context as defined in {{asym-msg2-proc}} and {{asym-msg3-proc}}, respectively.
 
 ## EDHOC-Exporter {#exporter}
 
 Application keys and other application specific data can be derived using the EDHOC-Exporter interface defined as:
 
 ~~~~~~~~~~~
    EDHOC-Exporter(label, context, length)
-     = EDHOC-KDF(PRK_out, '', label, context, length)
+     = EDHOC-KDF(PRK_exporter, label, context, length)
 ~~~~~~~~~~~
+where
 
-where label is a registered tstr from the EDHOC Exporter Label registry ({{exporter-label}}), context is a bstr defined by the application, and length is a uint defined by the application. The (label, context) pair must be unique, i.e., a (label, context) MUST NOT be used for two different purposes. However an application can re-derive the same key several times as long as it is done in a secure way. For example, in most encryption algorithms the same key can be reused with different nonces. The context can for example be the empty CBOR byte string.
+* label is a registered uint from the EDHOC Exporter Label registry ({{exporter-label}})
+* context is a bstr defined by the application
+* length is a uint defined by the application
 
-Examples of use of the EDHOC-Exporter are given in {{asym-msg4-proc}} and {{transfer}}.
+where
 
-* K_4 and IV_4 are derived with the EDHOC-Exporter using the empty CBOR byte string h'' as context, and labels "EDHOC_K_4" and "EDHOC_IV_4", respectively. IVs are only used if the EDHOC AEAD algorithm uses IVs.
+~~~~~~~~~~~~~~~~~~~~~~~
+PRK_exporter  = EDHOC-KDF( PRK_out, 10, h'', hash_length )
+~~~~~~~~~~~~~~~~~~~~~~~
+
+and where hash_length denotes the length of the hash function output in bytes, as specified by the COSE hash algorithm definition.
+
+The (label, context) pair must be unique, i.e., a (label, context) MUST NOT be used for two different purposes. However an application can re-derive the same key several times as long as it is done in a secure way. For example, in most encryption algorithms the same key can be reused with different nonces. The context can for example be the empty CBOR byte string.
+
+Examples of use of the EDHOC-Exporter are given in {{asym-msg4-proc}} and {{transfer}}.
```

Comment:
Should add some text that PRK_exporter needs to be recalculated after KeyUpdate.

Reply:
We also need more guidance on the use of KeyUpdate. But that might another PR.

Reply:
I've waited with this pending upcoming comment on key confirmation.